### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by glennrub**

### CODE REVIEW

File: nrf5/README.md [hunk 1]
```diff
@@ -12,35 +12,130 @@ This is a port of MicroPython to the Nordic nRF5 series of chips.
...
-  * [D52Q](https://www.dynastream.com/components/d52)
   * [Adafruit Feather nRF52](https://www.adafruit.com/product/3406)
+  * [Thingy:52](http://www.nordicsemi.com/eng/Products/Nordic-Thingy-52)
+  * [Arduino Primo](http://www.arduino.org/products/boards/arduino-primo)
+* nRF52840
+  * [PCA10056](http://www.nordicsemi.com/eng/Products/nRF52840-Preview-DK)
+
+## Compile and Flash
+
+Before any compilation of any targets some basic operatons needs to be done:
```
Comment (tralamazza) [inline nrf5/README.md]: Maybe reword it a bit: "Prerequisite steps for building the nrf5 port:"

File: nrf5/README.md [hunk 2]
```diff
@@ -12,35 +12,130 @@ This is a port of MicroPython to the Nordic nRF5 series of chips.
...
+
+    git clone <URL>.git micropython
+    cd micropython
+    git submodule update --init
+    make -C mpy-cross
+
+By default PCA10040 (nrf52832) is used as compile target. To build issue the following command inside the nrf5/ folder:
```
Comment (tralamazza) [inline nrf5/README.md]: maybe change to include flash: "... .To build and flash issue ..."

File: nrf5/README.md [hunk 3]
```diff
@@ -12,35 +12,130 @@ This is a port of MicroPython to the Nordic nRF5 series of chips.
...
+
+    make
+    make flash
+
+Alternativly the target board could be defined:
```
Comment (tralamazza) [inline nrf5/README.md]: typo: Alternatively

File: nrf5/README.md (hunk 1 above)
Comment (mattytrentini) [inline nrf5/README.md]: Typo: operations

**Comment by mattytrentini** [comment]
There's a lot more useful information in this updated README, and it's a big improvements - nice job!

There is still room for improvement though; because there are a number of boards supported I think it could be clearer to, in particular, separate out the various deployment processes. It's not that obvious that the Segger/PyOCD/DFU operations all perform similar roles but for different boards - and it's like that the user will only need one. Maybe they should be sub-headings under something like 'Deployment'? I'll try to come up with a quick example of what I mean before the end of the weekend...

**Comment by mattytrentini** [comment]
Oh! I remember that one of the minor issues I had with deploying to the Feather was that my USB device was named different to that in the makefile. I remember hacking the Makefile but before merging upstream we should allow it to be a configurable variable and document it in this README... (this is almost certainly the wrong place to document this!)

**Comment by glennrub** [comment]
Thanks a lot!

I will address your comment regarding the Serial port setting on the Feather52. I have the board now, and i was also a bit puzzled when it did not flash the first time. I have created #61 to follow this up before we go upstream. I'm suggesting to add "SERIAL" as a new makefile parameter.

File: nrf5/README.md (hunk 1 above)
Comment (glennrub) [reply nrf5/README.md]: addressed.

File: nrf5/README.md (hunk 2 above)
Comment (glennrub) [reply nrf5/README.md]: addressed.

File: nrf5/README.md (hunk 3 above)
Comment (glennrub) [reply nrf5/README.md]: addressed.

File: nrf5/README.md (hunk 1 above)
Comment (glennrub) [reply nrf5/README.md]: as text changed based on above comments, this is error is not present anymore.
